finagle-thrift/src/main/scala/com/twitter/finagle/Thrift.scala

```diff
@@ -100,15 +100,24 @@ object Thrift extends Client[ThriftClientRequest, Array[Byte]] with ThriftRichCl
     implicit object MaxReusableBufferSize extends Stack.Param[MaxReusableBufferSize] {
       val default = MaxReusableBufferSize(maxThriftBufferSize)
     }
+
+    /**
+     * A `Param` to control protocol upgrading.
```

Comment:
mind linking to the docs since this is a bit opaque. something like:

```
@see The [[https://twitter.github.io/finagle/guide/Protocols.html?highlight=Twitter-upgraded#thrift user guide]] for details on Twitter-upgrade Thrift.
```